File: 200-number-of-islands/number-of-islands.java
class Solution {

    void DFS(char[][] grid, int x, int y)
    {
        if(x < 0 || y < 0 || x == grid.length || y == grid[0].length || grid[x][y] == '0')
            return;
        
        grid[x][y] = '0';
        DFS(grid, x-1, y);
        DFS(grid, x+1, y);
        DFS(grid, x, y-1);
        DFS(grid, x, y+1);
    }
    public int numIslands(char[][] grid) {
        
        int count = 0;
        for(int i = 0 ; i < grid.length; i++)
        {
            for(int j = 0 ; j < grid[0].length ; j++)
            {
                if(grid[i][j] == '1')
                {
                    DFS(grid, i, j);
                    count++;
                }
            }
        }
        return count;
    }
}

File: 200-number-of-islands/number-of-islands.py
class Solution:

    def dfs(self, grid: List[List[str]], x , y) -> None:

        if x < 0 or y < 0 or x == len(grid) or y == len(grid[0]) or grid[x][y] == "0":
            return
        
        grid[x][y] = "0"
        self.dfs(grid, x-1, y)
        self.dfs(grid, x+1, y)
        self.dfs(grid, x, y-1)
        self.dfs(grid, x, y+1)

        return
            
    def numIslands(self, grid: List[List[str]]) -> int:

        n = len(grid)
        m = len(grid[0])
        ans = 0

        for i in range(n):
            for j in range(m):
                if(grid[i][j] == "1"):
                    ans+=1
                    self.dfs(grid, i, j)
        return ans
        